css/css-scroll-snap/scrollTo-scrollBy-snaps.html

```diff
@@ -74,7 +74,9 @@
 [
   [{left: 800}, 1000, 0],
   [{top: 900}, 0, 1000],
-  [{left: 900, top: 800}, 1000, 1000]
+  [{left: 900, top: 800}, 1000, 1000],
+  [{left: 800, top: -100}, 1000, 0], /* outside bounds on y axis */
+  [{left: 10000, top: -100}, 1000, 0] /* outside bounds on both axes */
```

Comment:
@majido, I am not sure here is the right place to comment about this PR, but anyways, using 10000px for scrollLeft doesn't sound right to me since the value is far from the target element, which means the target element (which is positioned at 1000px left) will not be visible in the snapport given by the scrollLeft (actually it will be clamped to scrollable area though) so that snapping shouldn't happen there?

Does this sound right? Or am I missing something?

Reply:
This is a fine place to comment. I will see these if you mention me. Also feel free to ping me on w3c #testing irc channel. I am also open to participate in other forums if it will help with communications around these tests to ensure we are testing scroll snap properly on both implementations.


As for the actual concern:

`left:10000` is out of bound so it basically (A) scrolls the scroller to its maximum extent then (B) we snap to the nearest target which is at 1000px.

You are that at then end of (A) the scroll position left=1000px is not visible. However, this does not mean we cannot snap to it. This was quite confusing to me initially as well and I asked for clarification which fantasai [provided](https://github.com/w3c/csswg-drafts/issues/2526#issuecomment-380371186): 

`In other words, https://drafts.csswg.org/css-scroll-snap/#snap-scope says that the state of being snapped requires that the thing you're snapped to must be visible. It doesn't say that you can't snap to a thing which is currently off-screen but would become on-screen after you have snapped to it.`

So by snapping to target at 1000px we satisfy both mandatory requirement and visibility requirement.